### PR TITLE
Fix check tests for cpprest, libpipeline, librsync, postgresql, libsoup

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix the %check tests for packages cpprest, libpipeline, librsync, postgresql. Also enables the %check for the libsoup package, which is working now that networking is enabled during test builds.

###### Change Log  

- Change specs to resolve %check issues for cpprest, libpipeline, librsync, postgresql
- Modify macros.override to enable libsoup check test

###### Does this affect the toolchain?  

YES
**NO**

###### Test Methodology

- Local package build

### CODE REVIEW

File: SPECS/libpipeline/libpipeline.spec [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+License:        GPLv3+
+URL:            http://libpipeline.nongnu.org
+Group:          Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
-Source0:	http://download.savannah.gnu.org/releases/libpipeline/%{name}-%{version}.tar.gz
-%define sha1 libpipeline=1c885e4d5551933c905e751048abb119593c53f4
+Source0:        http://download.savannah.gnu.org/releases/libpipeline/%{name}-%{version}.tar.gz
+
+BuildRequires:  check
```
Comment (PawelWMS) [inline SPECS/libpipeline/libpipeline.spec:12]: Could you surround this dependency with `%if %{with_check}`?

File: SPECS/librsync/librsync.spec [hunk 2]
```diff
@@ -11,6 +11,7 @@ Distribution:   Mariner
 Source0:        %{name}-%{version}.tar.gz
 
 BuildRequires:  cmake
+BuildRequires:  which
```
Comment (PawelWMS) [inline SPECS/librsync/librsync.spec:14]: Could you surround this dependency with `%if %{with_check}`?

File: SPECS/librsync/librsync.spec [hunk 3]
```diff
@@ -58,6 +59,9 @@ make %{?_smp_mflags}
 %install
 cd build
 make DESTDIR=%{buildroot} install
+# install outside of DESTDIR to resolve test error:
+# /usr/src/mariner/BUILD/librsync-2.0.2/build/rdiff: error while loading shared libraries: librsync.so.2: cannot open shared object file: No such file or directory
+make install
```
Comment (PawelWMS) [inline SPECS/librsync/librsync.spec:64]: We could also surround this step with `%if %{with_check}` to make the build faster when we're not running tests.

File: toolkit/resources/manifests/package/macros.override [hunk 4]
```diff
@@ -8,7 +8,6 @@
 # This list of skip_check_%{name} entries will cause %check to return immediately
 # on the specified spec.
 
-%skip_check_libsoup 1
```
Comment (PawelWMS) [inline toolkit/resources/manifests/package/macros.override:11]: Nice.:)

File: SPECS/librsync/librsync.spec (hunk 3 above)
Comment (anphel31) [reply SPECS/librsync/librsync.spec:68]: sure, done

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**